pkg/clustermesh/mcsapi/serviceimport_controller.go

```diff
@@ -0,0 +1,546 @@
+
+		conflictCount := 0
+		for _, currentSvcSpec := range svcExportByCluster {
+			if localField != getterStruct.getterFunc(currentSvcSpec) {
+				conflictCount += 1
+			}
+		}
+
+		if conflictCount > 0 {
+			return fmt.Sprintf(
+				"Conflicting %s. %d/%d clusters disagree. Using \"%s\" from oldest service export in cluster \"%s\".",
+				getterStruct.name, conflictCount, clusterCount, oldestField, oldestCluster,
+			)
+		}
+	}
+
+	return ""
+}
+
+func (r *mcsAPIServiceImportReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	nsExists, err := r.doesNamespaceExist(ctx, req)
+	if err != nil {
+		return controllerruntime.Fail(err)
+	}
+	if !nsExists {
+		return controllerruntime.Success()
+	}
+
+	svcImport, svcImportExists, err := r.getSvcImportBase(ctx, req)
+	if err != nil {
+		return controllerruntime.Fail(err)
+	}
+	svcExport, err := r.getSvcExport(ctx, req)
+	if err != nil {
+		return controllerruntime.Fail(err)
+	}
+
+	svcExportByCluster := r.globalServiceExports.GetServiceExportByCluster(req.NamespacedName)
+	if len(svcExportByCluster) == 0 && svcExport == nil {
+		if svcImportExists {
+			return controllerruntime.Fail(r.Client.Delete(ctx, svcImport))
+		}
+		return controllerruntime.Success()
+	}
+
+	if svcExport != nil {
+		localSvc, err := r.getLocalService(ctx, req)
+		if err != nil {
+			return controllerruntime.Fail(err)
+		}
+		if localSvc == nil {
+			if meta.SetStatusCondition(&svcExport.Status.Conditions, metav1.Condition{
+				Type:    mcsapiv1alpha1.ServiceExportValid,
+				Status:  metav1.ConditionFalse,
+				Reason:  "NoService",
+				Message: "Service doesn't exist",
+			}) {
+				meta.RemoveStatusCondition(&svcExport.Status.Conditions, mcsapiv1alpha1.ServiceExportConflict)
+				meta.RemoveStatusCondition(&svcExport.Status.Conditions, conditionTypeReady)
+				if err := r.Client.Status().Update(ctx, svcExport); err != nil {
+					return controllerruntime.Fail(err)
+				}
+			}
+			return controllerruntime.Success()
+		}
+		localSvcSpec := fromServiceToMCSAPIServiceSpec(localSvc, r.cluster, svcExport)
+		if svcExportByCluster == nil {
+			svcExportByCluster = operator.ServiceExportsByCluster{}
+		}
+		svcExportByCluster[r.cluster] = localSvcSpec
+
+		if localSvc.Spec.Type == corev1.ServiceTypeExternalName {
+			if meta.SetStatusCondition(&svcExport.Status.Conditions, metav1.Condition{
+				Type:    mcsapiv1alpha1.ServiceExportValid,
+				Status:  metav1.ConditionFalse,
+				Reason:  "ServiceType",
+				Message: "Service type ExternalName is not supported",
+			}) {
+				meta.RemoveStatusCondition(&svcExport.Status.Conditions, mcsapiv1alpha1.ServiceExportConflict)
+				meta.RemoveStatusCondition(&svcExport.Status.Conditions, conditionTypeReady)
+				if err := r.Client.Status().Update(ctx, svcExport); err != nil {
+					return controllerruntime.Fail(err)
+				}
+			}
+			return controllerruntime.Success()
+		}
+	}
+
+	oldestCluster := getOldestRemoteClusterService(svcExportByCluster)
```

Comment:
```suggestion
	oldestCluster := getOldestClusterService(svcExportByCluster)
```

Nit. The list of service exports may include the local one as well.